src/cloudai/_core/system.py

```diff
@@ -64,4 +63,75 @@ def __repr__(self) -> str:
 
     @abstractmethod
     def update(self) -> None:
-        raise NotImplementedError("Subclasses must implement this method.")
+        """
+        Update the system's state.
+
+        Raises
+            NotImplementedError: Raised if the method is not implemented in a subclass.
+        """
+        error_message = (
+            "System update method is not implemented. All subclasses of the System class must implement the "
+            "'update' method to ensure the system's state can be refreshed as needed."
+        )
+        logging.error(error_message)
+        raise NotImplementedError(error_message)
+
+    @abstractmethod
+    def is_job_running(self, job: "BaseJob") -> bool:
```

Comment:
Why do we need this function in `System` object? Can't users access `job` and ask its status?

Reply:
Yes, you are right; the base job class does not need to know about its system class. Previously, I added this to refactor the code for unifying the job spec generation interface. It turns out that we no longer need this.